hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -1784,10 +1788,12 @@ public UploadInfo putObject(PutObjectRequest putObjectRequest) {
    * @param putObjectRequest the request
    * @return the upload initiated
    * @throws AmazonClientException on problems
+   * @throws MetadataPersistenceException if metadata about the write could
+   * not be saved to the metadata store
```

Comment:
since this exception won't happen unless the config allows it, maybe reword to something like "could unacceptably not be saved to the metadata store" as a hint about that?

Reply:
updated